tests/foreman/ui/test_hostcollection.py

```diff
@@ -690,8 +690,119 @@ def test_positive_install_errata(self):
+               click on the link
+               "Change assigned Lifecycle Environment or Content View"
+            9. When a dialog box is open, select the new life cycle environment
+               and the new content view
+            10. Click on "Assign" button and click "Yes" button on confirmation
+                dialog when it appears
+            11. After last step the host collection change task page will
+                appear
+            12. Run "subscription-manager refresh" command on each host
+            13. Run "subscription-manager repos" command on each host
+
+        :expectedresults:
+            1. The host collection change task successfully finished
+            2. The "subscription-manager refresh" command successfully executed
+               and "All local data refreshed" message is displayed
+            3. The urls listed by last command "subscription-manager repos" was
+               updated to the new Life cycle environment and content view
+               names
+
+        :BZ: 1315280
+
+        :CaseLevel: System
+        """
+        # create a clients variable here as self.hosts attribute will be
+        # overwritten by Session initialization
+        clients = self.hosts
+        new_lce_name = gen_string('alpha')
+        new_cv_name = gen_string('alpha')
+        new_lce = entities.LifecycleEnvironment(
+            name=new_lce_name, organization=self.session_org).create()
+        new_content_view = entities.ContentView(
+            id=self.content_view.copy(data={u'name': new_cv_name})['id']
+        )
+        new_content_view.publish()
+        new_content_view = new_content_view.read()
+        new_content_view_version = new_content_view.version[0]
+        new_content_view_version.promote(data={'environment_id': new_lce.id})
+        # repository urls listed by command "subscription-manager repos" looks
+        # like:
+        # Repo URL  : https://{host}/pulp/repos/{org}/{lce}/{cv}/custom
+        # /{product_name}/{repo_name}
+        # changing lce and content view should affect only the {lce}/{cv} part
+        repo_line_start_with = 'Repo URL:  '
+        # the lce and cv location part of the current repos urls
+        env_cv_repo_location = '{0}/{1}'.format(
+            self.env.name, self.content_view.name)
+        # the expected lce and cv location part of the expected repos urls when
+        # new lce and cv will be assigned
+        new_env_cv_repo_location = '{0}/{1}'.format(
+            new_lce_name, new_cv_name)
+        expected_repo_urls = {client.hostname: [] for client in clients}
+        for client in clients:
+            result = client.run("subscription-manager repos")
+            self.assertEqual(result.return_code, 0)
+            client_repo_urls = [
+                line.split(' ')[-1]
+                for line in result.stdout
+                if line.startswith(repo_line_start_with)
+            ]
+            self.assertGreater(len(client_repo_urls), 0)
```

Comment:
maybe we can change that assertion on:
```
self.assertEqual(
    set(expected_repo_urls[client.hostname]),
    set(client_repo_urls)
)
```
just to have better visibility that setting was really changed?

Reply:
@oshtaier these sets cannot be equal as the expected URLs has been updated by the new lce/cv values, perhaps you mean assertNotEqual?

Reply:
@oshtaier any think we need to be sure that there some urls there and add the assert not equal after expected_repo_urls initialization after line 779

Reply:
I meant specifically to check before update and after